web3/contract.py

```diff
@@ -351,7 +351,7 @@ def encodeABI(cls, fn_name, args=None, kwargs=None, data=None):
         :param data: defaults to function selector
         """
         fn_abi, fn_selector, fn_arguments = get_function_info(
-            fn_name, contract_abi=cls.abi, args=args, kwargs=kwargs,
+            fn_name, cls.web3.codec, contract_abi=cls.abi, args=args, kwargs=kwargs,
```

Comment:
just noticing that we still call it `web3` instead of `w3` here.  Might be a good future cleanup to do.